pkg/core/interop_neo_test.go

```diff
@@ -288,6 +289,122 @@ func TestECDSAVerify(t *testing.T) {
 	})
 }
 
+func initCHECKMULTISIG(msg []byte, n int) ([]vm.StackItem, []vm.StackItem, map[string]*keys.PublicKey, error) {
+	var err error
+
+	keyMap := make(map[string]*keys.PublicKey)
+	pkeys := make([]*keys.PrivateKey, n)
+	pubs := make([]vm.StackItem, n)
+	for i := range pubs {
+		pkeys[i], err = keys.NewPrivateKey()
+		if err != nil {
+			return nil, nil, nil, err
+		}
+
+		pk := pkeys[i].PublicKey()
+		data := pk.Bytes()
+		pubs[i] = vm.NewByteArrayItem(data)
+		keyMap[string(data)] = pk
+	}
+
+	sigs := make([]vm.StackItem, n)
+	for i := range sigs {
+		sig := pkeys[i].Sign(msg)
+		sigs[i] = vm.NewByteArrayItem(sig)
+	}
+
+	return pubs, sigs, keyMap, nil
+}
+
+func subSlice(arr []vm.StackItem, indices []int) []vm.StackItem {
+	if indices == nil {
+		return arr
+	}
+
+	result := make([]vm.StackItem, len(indices))
+	for i, j := range indices {
+		if j >= len(arr) {
+			result[i] = vm.NewByteArrayItem([]byte(fmt.Sprintf("invalid pkey %d", j)))
+		} else {
+			result[i] = arr[j]
+		}
+	}
+
+	return result
+}
+
+func initCHECKMULTISIGVM(t *testing.T, n int, ik, is []int) *vm.VM {
+	v := vm.New()
+
+	msg := []byte("NEO - An Open Network For Smart Economy")
+
+	pubs, sigs, _, err := initCHECKMULTISIG(msg, n)
+	require.NoError(t, err)
+
+	pubs = subSlice(pubs, ik)
+	sigs = subSlice(sigs, is)
+
+	v.Estack().PushVal(sigs)
+	v.Estack().PushVal(pubs)
+	v.Estack().PushVal(msg)
+
+	return v
+}
+
+type testParam struct {
+	ic    *interopContext
+	isErr bool
+	res   bool
+}
+
+func multisigTest(t *testing.T, p *testParam, n int, ik, is []int) {
+	v := initCHECKMULTISIGVM(t, n, ik, is)
+
+	err := p.ic.ecdsaCheckMultisig(v)
+	if p.isErr {
+		require.Error(t, err)
+		return
+	}
+
+	require.NoError(t, err)
+	require.Equal(t, 1, v.Estack().Len())
+	require.Equal(t, p.res, v.Estack().Pop().Bool())
+}
+
+func TestECDSACheckMultisig(t *testing.T) {
```

Comment:
Same thing with multisig-specific tests, duplicating them makes no sense, we either should have them in one place, or in the other.

Reply:
Fixed.